pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -156,6 +158,34 @@ func GetEnvironment(cr *api.KieApp, service kubernetes.PlatformService) (api.Env
 	return mergedEnv, nil
 }
 
+func handleAdminSecretUsernameAndPassword(cr *api.KieApp, service kubernetes.PlatformService) {
+	if len(cr.Spec.CommonConfig.SecretAdminCredentials) > 0 && len(cr.Spec.CommonConfig.AdminPassword) > 0 {
+		// AdminUsername is always set from the cr or with default value
+		// we check if the secret exists,
+		_, err := getSecret(service, cr.Namespace, cr.Spec.CommonConfig.SecretAdminCredentials)
+		if err != nil {
+			// we create the secret with the credentials provided
+			errCreateSecret := createSecret(service, cr.Spec.CommonConfig.SecretAdminCredentials, cr.Spec.CommonConfig.AdminUser, cr.Spec.CommonConfig.AdminPassword, cr)
+			if errCreateSecret != nil {
+				log.Error("Can't create Admin Secret with the provided secret name and credentials", errCreateSecret)
+			}
+		}
+	} else if len(cr.Spec.CommonConfig.SecretAdminCredentials) > 0 && len(cr.Spec.CommonConfig.AdminPassword) == 0 {
```

Comment:
Can we combine these to if block to one? We have there basically code duplicates. First we are trying to get the secret and then create a one if not exists with given credentials

Reply:
ok